adapters/handlers/rest/handlers_authz.go

```diff
@@ -52,24 +52,33 @@ func setupAuthZHandlers(api *operations.WeaviateAPI, controller authorization.Co
 }
 
 func (h *authZHandlers) createRole(params authz.CreateRoleParams, principal *models.Principal) middleware.Responder {
-	// TODO validate and audit log
 	if err := h.authorizer.Authorize(principal, authorization.CREATE, authorization.Roles()...); err != nil {
 		return authz.NewCreateRoleForbidden().WithPayload(errPayloadFromSingleErr(err))
 	}
 
 	if *params.Body.Name == "" {
-		return authz.NewCreateRoleUnprocessableEntity().WithPayload(errPayloadFromSingleErr(errors.New("role name is required")))
+		return authz.NewCreateRoleBadRequest().WithPayload(errPayloadFromSingleErr(errors.New("role name is required")))
+	}
+
+	if len(params.Body.Permissions) == 0 {
+		return authz.NewCreateRoleBadRequest().WithPayload(errPayloadFromSingleErr(errors.New("role has to have at least 1 permission")))
 	}
 
 	if slices.Contains(rbac.BuiltInRoles, *params.Body.Name) {
 		return authz.NewCreateRoleForbidden().WithPayload(errPayloadFromSingleErr(fmt.Errorf("you can not create role with the same name as builtin role %s", *params.Body.Name)))
 	}
```

Comment:
**Issue (non-blocking):** was going to leave this comment to another PR on this clause but I see that you've added mock tests for this exact error so I'll add here. lmk wyt 😁 

Should this really be a `403`? Since we have already authorized the request, the user does have permission to attempt the request so returning `403` for this specific error might be confusing. Would it not be better to have it as a `400` because the request is denied based on the content of its body rather than its authorization info?